webpack/assets/javascripts/react_app/redux/reducers/hosts/storage/vmware.js

```diff
@@ -26,7 +26,18 @@ const initialState = Immutable({
   volumes: [],
 });
 
-const availableControllerKeys = [1000, 1001, 1002, 1003, 1004];
+const availableControllerKeys = [
+  1000,
```

Comment:
Can you please give me a bit more details about this array and the meaning of the items in it?

Reply:
This is to assign the keys to scsi or nvme controllers. We can have max of 4 controllers each so the total number of available keys are 8.

Reply:
When I was in university my professor would reduce your grade if you used magic values. Anything other than 0, 1 or 2 needed to be a constant that described what the magic value actually was.

In Foreman that rule isn't strictly applied, but another way to write this would be:
```javascript
Array.from({length: 8}, (value, index) => 1000 + index);
```

I'm not sure if it's that much clearer, but as I read the code it's a list of 8 numbers where the base number is 1000.

Having said that, how do you prevent a user from assigning 8 nvme controllers or 8 scsi controllers?